scanner/mappers/mappers.go

```diff
@@ -788,3 +788,64 @@ func findName(vuln *claircore.Vulnerability, p *regexp.Regexp) (string, bool) {
 	}
 	return "", false
 }
+
+// filterVulns filters repeat vulnerabilities out of vulnIDs and returns the result.
+// This function does not guarantee ordering is preserved.
+func filterVulns(vulnIDs []string, vulns map[string]*claircore.Vulnerability) []string {
+	// Group each vulnerability by name.
+	// This maps each name to a slice of vulnerabilities to protect against the possibility
+	// Claircore finds multiple vulnerabilities with the same name for this package from different vulnerability streams.
+	// In that situation, it is not clear which one may be the single, correct option to choose, so just allow for both.
+	vulnsByName := make(map[string][]*claircore.Vulnerability)
+OUTER:
+	for _, vulnID := range vulnIDs {
+		vuln := vulns[vulnID]
+		if vuln == nil {
+			continue
+		}
+
+		// Find the currently tracked vulnerabilities with the same name.
+		// If this entry matches any of those, then ignore this one.
+		matching := vulnsByName[vuln.Name]
+		for _, match := range matching {
+			if vulnsEqual(match, vuln) {
+				continue OUTER
+			}
+		}
+
+		// Add the unique entry to the map.
+		vulnsByName[vuln.Name] = append(vulnsByName[vuln.Name], vuln)
+	}
+
+	filtered := make([]string, 0, len(vulns))
```

Comment:
This will allocate a slice of the size of all vulnerabilities in the report per package.